common/app/views/fragments/atoms/media.scala.html

```diff
@@ -1,14 +1,9 @@
-@import views.support.RenderClasses
-
-@(media: model.content.MediaAtom)(implicit request: RequestHeader)
-
-<div class="@RenderClasses(Map(
-    "u-responsive-ratio" -> true,
-    "u-responsive-ratio--hd" -> true
-))">
-
-@Html(media.defaultHtml)
-
-</div>
+@(media: _root_.model.content.MediaAtom)(implicit request: RequestHeader)
 
+@{
+    media match {
+            case youtube if media.assets.head.platform == "Youtube" => views.html.fragments.atoms.youtube(media)
+            case _ => views.html.fragments.atoms.genericMedia(media)
+        }
```

Comment:
I know we do it all over the codebase but logic in templates makes refactoring very hairy.
If you cannot do it otherwise fine but I would prefer to avoid it if possible